src/main/scala/com/cognite/sdk/scala/v1/resources/dataModels.scala

```diff
@@ -60,28 +56,78 @@ class DataModels[F[_]](val requestSession: RequestSession[F])
 }
 
 object DataModels {
-  implicit val dataModelPropertyIndexEncoder: Encoder[DataModelPropertyIndex] =
-    deriveEncoder[DataModelPropertyIndex]
+  implicit val dataModelIdentifierEncoder: Encoder[DataModelIdentifier] =
+    Encoder
+      .encodeIterable[String, Seq]
+      .contramap[DataModelIdentifier](dmi =>
+        dmi match {
+          case DataModelIdentifier(Some(space), model) =>
+            Seq(space, model)
+          case DataModelIdentifier(_, model) =>
+            Seq(model)
+        }
+      )
+
+  implicit val bTreeIndexEncoder: Encoder[BTreeIndex] =
+    deriveEncoder[BTreeIndex]
+  implicit val dataModelPropertyTypeEncoder: Encoder[PropertyType] =
+    Encoder.encodeString.contramap[PropertyType](_.code)
+  implicit val dataModelPropertyIndexesEncoder: Encoder[DataModelIndexes] =
+    deriveEncoder[DataModelIndexes]
   implicit val dataModelPropertyEncoder: Encoder[DataModelProperty] =
     deriveEncoder[DataModelProperty]
-  implicit val dataModelEncoder: Encoder[DataModel] = deriveEncoder[DataModel]
-  implicit val dataModelItemsEncoder: Encoder[Items[DataModel]] = deriveEncoder[Items[DataModel]]
+  implicit val contstrainedPropertyEncoder: Encoder[ContstrainedProperty] =
+    deriveEncoder[ContstrainedProperty]
+  implicit val uniquenessConstraintEncoder: Encoder[UniquenessConstraint] =
+    deriveEncoder[UniquenessConstraint]
+  implicit val dataModelConstraintsEncoder: Encoder[DataModelConstraints] =
+    deriveEncoder[DataModelConstraints]
+  implicit val dataModelEncoder: Encoder[DataModel] =
+    deriveEncoder[DataModelDTO].contramap[DataModel](_.toDTO)
+  implicit val dataModelItemsEncoder: Encoder[SpacedItems[DataModel]] =
+    deriveEncoder[SpacedItems[DataModel]]
+  implicit val cogniteIdSpacedItemsEncoder: Encoder[SpacedItems[CogniteId]] =
+    deriveEncoder[SpacedItems[CogniteId]]
   implicit val dataModelListInputEncoder: Encoder[DataModelListInput] =
     deriveEncoder[DataModelListInput]
 
-  implicit val dataModelPropertyIndexDecoder: Decoder[DataModelPropertyIndex] =
-    deriveDecoder[DataModelPropertyIndex]
+  implicit val dataModelIdentifierDecoder: Decoder[DataModelIdentifier] =
+    Decoder
+      .decodeIterable[String, List]
+      .map(ids =>
+        ids match {
```

Comment:
Also neat trick (which IntelliJ should tell you): You can just get rid of the 
```
ids => ids match
```
here as Scala will understand what you're trying to do (`.map(case x =>..) `